### DISCUSSION TITLE
dns.resolve documentation confusion

### DISCUSSION THREAD

**Discussion by menocomp** [nodejs/help#4983] [Q&A]
### Affected URL(s)

https://nodejs.org/en/docs/meta/topics/dependencies/#c-ares, https://nodejs.org/api/dns.html#dns_dns_resolve_dns_resolve_and_dns_reverse

### Description of the problem

Both of the documents I shared above refer to the usage of `dns.resolve`.

The first document: https://nodejs.org/en/docs/meta/topics/dependencies/#c-ares says: `c-ares supports /etc/hosts`
However the second document https://nodejs.org/api/dns.html#dns_dns_resolve_dns_resolve_and_dns_reverse says: `they do not use the configuration from /etc/hosts`

Which one is right here?

**Comment by cjihrig** [comment]
I believe what those two docs are saying is that the c-ares library has APIs that are capable of `/etc/hosts` lookups, but the c-ares APIs exposed via Node don't utilize `/etc/hosts`.

**Reply by menocomp** [reply]
@cjihrig Thanks for your quick reply. I agree with you on that. 

Shouldn't the documentation be updated to reflect that? as it is currently a bit confusing.

**Reply by cjihrig** [reply]
Do you want to send a pull request to make the docs more clear?

**Reply by menocomp** [reply]
Sure. It would be be nice.

It would be first time for me to create PR for Node documentation. Do you suggest any documentation for me to read first before creating PR?

**Reply by cjihrig** [reply]
You could take a look at this guide: https://github.com/nodejs/node/blob/main/doc/contributing/pull-requests.md

That page is also linked to from the more general [CONTRIBUTING.md](https://github.com/nodejs/node/blob/main/CONTRIBUTING.md) file in the root of the project, which is probably more discoverable.
